queries/d/highlights.scm

```diff
@@ -35,254 +86,194 @@
+  (switch)
 ] @conditional
 
 [
-  "break"
-  "continue"
-  "do"
-  "for"
-  "foreach"
-  "foreach_reverse"
-  "while"
+  (break)
+  (continue)
+  (do)
+  (for)
+  (while)
+  (foreach)
+  (foreach_reverse)
 ] @repeat
 
 [
-  "__parameters"
-  "alias"
-  "align"
-  "asm"
-  "auto"
-  "body"
-  "class"
-  "debug"
-  "enum"
-  "export"
-  "goto"
-  "interface"
-  "invariant"
-  "macro"
-  "out"
-  "override"
-  "package"
-  "static"
-  "struct"
-  "template"
-  "union"
-  "unittest"
-  "version"
-  "with"
-] @keyword
-
-[
-  "delegate"
-  "function"
+  (delegate)
+  (function)
 ] @keyword.function
 
-"return" @keyword.return
+(return) @keyword.return
 
-[
-  "cast"
-  "new"
-] @keyword.operator
+(debug) @debug
+(import) @include
 
 [
-  "+"
-  "++"
-  "+="
-  "-"
-  "--"
-  "-="
-  "*"
-  "*="
-  "%"
-  "%="
-  "^"
-  "^="
-  "^^"
-  "^^="
-  "/"
-  "/="
-  "|"
-  "|="
-  "||"
-  "~"
-  "~="
-  "="
-  "=="
-  "=>"
-  "<"
-  "<="
-  "<<"
-  "<<="
-  ">"
-  ">="
-  ">>"
-  ">>="
-  ">>>"
-  ">>>="
-  "!"
-  "!="
-  "&"
-  "&&"
-] @operator
-
-[
-  "catch"
-  "finally"
-  "throw"
-  "try"
-] @exception
-
-"null" @constant.builtin
-
-[
-  "__gshared"
-  "const"
-  "immutable"
-  "shared"
-] @storageclass
+  (abstract)
+  (alias)
+  (align)
+  (asm)
+  (assert)
+  (cast)
+  (class)
+  (delete)
+  (enum)
+  (export)
+  (extern)
+  (final)
+  (in)
+  (inout)
+  (interface)
+  (invariant)
+  (is)
+  (lazy)
+  ; "macro" - obsolete
+  (mixin)
+  (module)
+  (new)
+  (nothrow)
+  (out)
+  (override)
+  (package)
+  (pragma)
+  (private)
+  (protected)
+  (public)
+  (pure)
+  (ref)
+  (static)
+  (struct)
+  (super)
+  (synchronized)
+  (template)
+  (this)
+  (throw)
+  (typeid)
+  (typeof)
+  (union)
+  (unittest)
+  (version)
+  (with)
+  (gshared)
+  (traits)
+  (vector)
+  (parameters_)
+] @keyword
 
-[
-  "abstract"
-  "deprecated"
-  "extern"
-  "final"
-  "inout"
-  "lazy"
-  "nothrow"
-  "private"
-  "protected"
-  "public"
-  "pure"
-  "ref"
-  "scope"
-  "synchronized"
-] @type.qualifier
+(label (identifier) @label)
+(goto_statement (goto) @keyword (identifier) @label)
 
-(alias_assignment
-  . (identifier) @type.definition)
+(at_attribute) @attribute
 
-(module_declaration
-  "module" @include
-)
+;; Literals
 
-(import_declaration
-  "import" @include
-)
+(int_literal) @number
+(float_literal) @float
+(char_literal) @character
+(string_literal) @string
 
-(type) @type
+[
+  (true)
+  (false)
+] @boolean
 
-(catch_parameter
-  (qualified_identifier) @type
-)
+;; Constants
 
-(var_declarations
-  (qualified_identifier) @type
-)
+(null) @constant.builtin
+(special_keyword) @constant.builtin
 
-(func_declaration
-  (qualified_identifier) @type
-)
+;; Functions
 
-(parameter
-  (qualified_identifier) @type
-)
+(function_declaration (identifier) @function)
 
-(class_declaration
-  (identifier) @type
-)
+(call_expression (identifier) @function.call)
+(call_expression (type (identifier) @function.call))
 
-(fundamental_type) @type.builtin
+;; Types
 
```

Comment:
apparently `(type (identifier) @type)` is missing. I don't see any type highlights and also couldn't find `@type` in this `highlighs.scm`

Reply:
strangely a function name in a call is also a `(type (identifier) @type)`

Reply:
Grammatically in D -- the distinction between a function call and a type is not something you can easily distinguish using syntax alone in many circumstances.   (I don't love D's grammar, but I did the tree-sitter plugin because I'm forced to work in it at work.)

Reply:
Done

Reply:
I'm still not able to get types highlighted :/